src/llm/apis/openai_completions.hpp

```diff
@@ -0,0 +1,125 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <optional>
+#include <set>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <openvino/genai/generation_config.hpp>
+#include <openvino/genai/generation_handle.hpp>
+#include <rapidjson/document.h>
+
+#include "absl/status/status.h"
+
+using namespace rapidjson;
+
+namespace ovms {
+
+struct StreamOptions {
+    bool includeUsage = false;
+};
+
+using chat_entry_t = std::unordered_map<std::string, std::string>;
+using chat_t = std::vector<chat_entry_t>;
+
+#define IGNORE_EOS_MAX_TOKENS_LIMIT 4000
+
+// Class that maps OpenAI request content and provides methods to create GenerationConfig from it.
+// It's supposed to be used only by OpenAIChatCompletionsHandler class
+struct OpenAIChatCompletionsRequest {
```

Comment:
Maybe we could hide this structure in openai_completions.cpp since now nothing really uses it besides the handler